ui/Screen/Project/Source.svelte

```diff
@@ -45,6 +47,19 @@
       type: path.extractProjectSourceObjectType($location)
     });
   }
+
+  let readmePath = null;
+
+  $: readme = null;
+  $: if ($object.status === remote.Status.Success) {
+    if ($object.data.info.objectType === TREE) {
+      readmePath = findReadme($object.data);
+
+      if (path) {
+        readme = blob($project.data.id, $currentRevision, readmePath);
+      }
+    }
+  }
```

Comment:
Not sure how to get rid of this spaghetti code: I essentially need to chain the `$object` fetching with the `$readme` fetching.

Reply:
Can you merge master, your `Source.svelte` is quite out-of-date.